main.go

```diff
@@ -67,14 +70,30 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+func runPrometheusServer(metricsAddr string) {
+	setupLog.Info("Entered runPrometheusServer")
+	mux := restful.NewContainer()
+	mux.Handle("/metrics", promhttp.Handler())
+	setupLog.Info("Defining Prometheus metrics endpoint server")
```

Comment:
I think this log output and the one above are unnecessary. They would probably not be useful for debugging.

If you want, you can move the log command from line 81, to the top of the function.

Reply:
ok left just one log entry